src/utils/tokens.ts

```diff
@@ -19,7 +20,18 @@ export const getERC20TokenInfoOnChain = async (
   if (!web3) return
 
   const erc20 = ERC20__factory.connect(address, web3)
-  const [symbol, decimals] = await Promise.all([erc20.symbol(), erc20.decimals()])
+
+  const symbol = await erc20
+    .symbol()
+    .then((symbol) => symbol)
+    .catch((error) => parseBytes32String(error.value)) // Some older contracts use bytes32 instead of string
+    .finally(() => '')
+
+  const decimals = await erc20
+    .decimals()
+    .then((decimals) => decimals)
+    .catch(() => BigInt(18))
```

Comment:
This could potentially lead to wrong decimals. Does the contract also use a exotic decimal function?

Reply:
Not that I know of. I will revert this for the decimals